src/controllers/urls.go

```diff
@@ -1 +1,52 @@
 package controllers
+
+import (
+	"errors"
+	"gorm.io/gorm"
+	"onepixel_backend/src/helpers"
+	"onepixel_backend/src/models"
+)
+
+type UrlController struct {
+	// db
+	db *gorm.DB
+}
+
+func NewUrlController(db *gorm.DB) *UrlController {
+	return &UrlController{
+		db: db,
+	}
+}
+
+// Create new url
+func (c *UrlController) Create(longUrl string, groupId uint, userId uint, iteration int) (*models.Url, error) {
+	if iteration == 10 {
+		return nil, errors.New("cannot create a new URL")
```

Comment:
for first draft, let's not do the recursion, and simply error if key is invalid. let's client try again